### PR TITLE
Feat(eos_designs): Loopbacks data model under tenant vrfs

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

Adds data model for loopback_interfaces under tenant vrfs similar to l3_interfaces.

```yaml
...
    vrfs:
      - name: <name>
        loopbacks:
          - loopback: <int> # Ex. 123
            ip_address: <ip/mask>
            node: <str>
            description: <str>
            ospf:
              enabled: <bool>
              area: <str> # Ex. 0.0.0.0 or 0
            raw_eos_cli: |
              <multiline cli>
```

## Component(s) name

`arista.avd.eos_designs`

## How to test

Tested with molecule scenario

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
I would prefer just to have one ip address, interface, node per entry.

**Comment by emilarista** [comment]
> I would prefer just to have one ip address, interface, node per entry.

Changed the data model according to this.

**Review by philippebureau — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/virtual_source_nat_vrfs.py [hunk 1]
```diff
@@ -33,12 +33,17 @@ def virtual_source_nat_vrfs(self) -> list | None:
...
+            for vrf in virtual_source_nat_vrfs:
+                if loopback_interface["vrf"] == vrf["name"]:
+                    vrf_exists = True
+            if not vrf_exists:
+                virtual_source_nat_vrfs.append(
+                    {
+                        "name": loopback_interface["vrf"],
+                        "ip_address": loopback_interface["ip_address"].split("/", maxsplit=1)[0],
+                    }
+                )
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/virtual_source_nat_vrfs.py]: As discussed we need to find a better way for this logic. I will look into it.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Comment by ClausHolbechArista** [comment]
@philippebureau I modified this PR a bit since your review. Can you take another round? Notice the schema changed slightly. Thanks.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by ankudinov — COMMENTED** [review]
Tested the PR in Codespace (adjust devcontainer.json to the new image if you want to reproduce), AVD examples single-dc inventory.

Following `/workspaces/ansible-avd/ansible_collections/arista/avd/examples/single-dc-l3ls/group_vars/NETWORK_SERVICES.yml` produces confusing results:

```yaml
---
tenants:
  # Definition of tenants. Additional level of abstraction to VRFs
  - name: TENANT1
    # Number used to generate the VNI of each VLAN by adding the VLAN number in this tenant.
    mac_vrf_vni_base: 10000
    vrfs:
      # VRF definitions inside the tenant.
      - name: VRF10
        # testing https://github.com/aristanetworks/avd/pull/3486
        loopbacks:
          - loopback: 1234
            ip_address: 1780.10.10.30/32
            node: dc1-leaf1a
            description: just-a-test-loopback
            raw_eos_cli:
              description description-from-raw-eos-cli
          - loopback: 2345
            ip_address: 192.10.10.20/234
            node: dc1-leaf1b
            description: test2345
          - loopback: 987
            ip_address: 192.30.30.20/32
            node: dc1-leaff1b
            description: test2345
        # VRF VNI definition.
        vrf_vni: 10
        # Enable VTEP Network diagnostics
        # This will create a loopback with virtual source-nat enable to perform diagnostics from the switch.
        vtep_diagnostic:
          # Loopback interface number
          loopback: 10
          # Loopback ip range, a unique ip is derived from this ranged and assigned
          # to each l3 leaf based on it's unique id.
          loopback_ip_range: 10.255.10.0/27
        svis:
          # SVI definitions.
          - id: 11
            # SVI Description
            name: VRF10_VLAN11
            enabled: true
            # IP anycast gateway to be used in the SVI in every leaf.
            ip_address_virtual: 10.10.11.1/24
          - id: 12
            name: VRF10_VLAN12
            enabled: true
            ip_address_virtual: 10.10.12.1/24
      - name: VRF11
        loopbacks:
          - loopback: 4321
            ip_address: 1780.10.10.30/32
            node: dc1-leaf2a
            description: just-a-test-loopback
        vrf_vni: 11
        vtep_diagnostic:
          loopback: 11
          loopback_ip_range: 10.255.11.0/27
        svis:
          - id: 21
            name: VRF11_VLAN21
            enabled: true
            ip_address_virtual: 10.10.21.1/24
          - id: 22
            name: VRF11_VLAN22
            enabled: true
            ip_address_virtual: 10.10.22.1/24

    l2vlans:
      # These are pure L2 vlans. They do not have a SVI defined in the l3leafs and they will be bridged inside the VXLAN fabric
      - id: 3401
        name: L2_VLAN3401
      - id: 3402
        name: L2_VLAN3402
```

1. There is no verification for IP address format. `1780.10.10.30/32` is accepted and inserted in the config as well as `192.10.10.20/234`. Not sure about complexity and if we should plan IP address validation at this stage, but it's feasible to list that somewhere in limitations for the data model.
2. Creating a loopback on non-existing leaf fails silently. Typing hostnames incorrectly can be quite common in the field and this should be detected to avoid hunting that in future inventories.
3. I'd consider switching `node` and `ip_address` to list. As the case of provisioning same loopback for a similar purpose on multiple switches can be quite common.

If required - happy to LGTM and park in a different issue as the PR works for me in general.

**Comment by ClausHolbechArista** [comment]
> There is no verification for IP address format. 1780.10.10.30/32 is accepted and inserted in the config as well as 192.10.10.20/234. Not sure about complexity and if we should plan IP address validation at this stage, but it's feasible to list that somewhere in limitations for the data model.

We don't have that schema support yet. But schema descriptions should mention for expected format until then.

> Creating a loopback on non-existing leaf fails silently. Typing hostnames incorrectly can be quite common in the field and this should be detected to avoid hunting that in future inventories.

That is the same everywhere in AVD today. It would be quite "expensive" to perform validation of these fields in a scaled setup.

> I'd consider switching node and ip_address to list. As the case of provisioning same loopback for a similar purpose on multiple switches can be quite common.

If you check the history of this PR, it started as lists. Since we already have a model for common loopbacks, this is more thought as a form of override for special cases. Also we would like to avoid more models with multiple lists that the user has to align.


Next review, please comment in the code/schema as separate points so we can track/reply/resolve individually. These large comments makes it harder to follow.

**Review by ankudinov — APPROVED** [review]
LGTM

**Pull request merged**